omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptAttributeTypeController2_0Test.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.controller.openmrs2_0;
+
+import org.apache.commons.beanutils.PropertyUtils;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.api.ConceptService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.web.v1_0.RestTestConstants2_0;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+/**
+ * Tests functionality of {@link ConceptAttributeTypeController}.
```

Comment:
Tests the functionality of what?

Reply:
class not yet provided.... provided as guidance

Reply:
should I remove it @dkayiwa

Reply:
Which class are you providing this test for?

Reply:
`Tests functionality of {@link ConceptAttributeTypeController}.` **have done this due to the system used here**
>
/**
 * Tests functionality of {@link ConceptAttributeController}.
 */
public class ConceptAttributeController2_0Test extends MainResourceControllerTest {

**Which yet**  `ConceptAttributeController` **is not also found**

Reply:
Aren't you testing for the domain object concept attribute type?

Reply:
Am not getting you when you say domain Object but According to [Platform 2.4.0 Restws endpoint doc](https://docs.google.com/spreadsheets/d/1vrVYIjVZqxzS2FHmD63KGV2wqvpJN-x-k77atS6TtIM/edit#gid=0) line 61-63.....
Shows according to the` Url` provided that this is the way to go like when you look at the` URL ``http://localhost:8087/openmrs/ws/rest/v1/conceptattributetype`

Reply:
Let me ask it differently, what is this test all about?

Reply:
Oh the test is about.....to `Create The ConceptAtrributeType` which to create it you need a URL `http://localhost:8087(servername)/openmrs/ws/rest/v1/conceptattributetype` otherwise using `ConceptAttributeController2_0Test` wouldnot work since its URL is `http://localhost:8087(servername)/openmrs/ws/rest/v1/concept/{Uuid}/attribute`

Reply:
So you are not testing the controller.

Reply:
Yah ..... but I need a new class to implement `http://localhost:8087(servername)/openmrs/ws/rest/v1/conceptattributetype` not so?

if so removing this
>/**
 * Tests functionality of {@link ConceptAttributeTypeController}.

I think will solve this out 🤔

Reply:
cc @dkayiwa